### PR TITLE
store/helper: use random server port in test

### LINKED ISSUE #14986: unit test HelperTestSuite fails
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.

```
[2020-02-27T12:22:52.592Z] ----------------------------------------------------------------------
[2020-02-27T12:22:52.592Z] FAIL: helper_test.go:63: HelperTestSuite.SetUpSuite
[2020-02-27T12:22:52.592Z] 
[2020-02-27T12:22:52.592Z] helper_test.go:136:
[2020-02-27T12:22:52.592Z]     c.Assert(err, IsNil)
[2020-02-27T12:22:52.592Z] ... value *net.OpError = &net.OpError{Op:"listen", Net:"tcp", Source:net.Addr(nil), Addr:(*net.TCPAddr)(0xc000341c80), Err:(*os.SyscallError)(0xc00035f5c0)} ("listen tcp 127.0.0.1:10100: bind: address already in use")
[2020-02-27T12:22:52.592Z] 
[2020-02-27T12:22:52.592Z] 
[2020-02-27T12:22:52.592Z] ----------------------------------------------------------------------
```
2. What did you expect to see?

Listen to a random port.

3. What did you see instead?

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?

### PR DESCRIPTION
**Pull request by SunRunAway**
### What problem does this PR solve? 

Fixes #14986

### What is changed and how it works?

Use random server port
### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: store/helper/helper_test.go [hunk 1]
```diff
@@ -61,13 +62,13 @@ func (s *mockStore) TLSConfig() *tls.Config {
...
 func (s *HelperTestSuite) SetUpSuite(c *C) {
-	go s.mockPDHTTPServer(c)
+	url := s.mockPDHTTPServer(c)
 	time.Sleep(100 * time.Millisecond)
 	mvccStore := mocktikv.MustNewMVCCStore()
 	mockTikvStore, err := mockstore.NewMockTikvStore(mockstore.WithMVCCStore(mvccStore))
 	s.store = &mockStore{
 		mockTikvStore.(tikv.Storage),
-		[]string{"127.0.0.1:10100/"},
+		[]string{url[len("http://"):]},
```
Comment (zz-jason) [inline store/helper/helper_test.go]: would it use the port that already in use?

Comment (SunRunAway) [reply store/helper/helper_test.go]: No, `httptest.NewServer` will binds the socket to port 0, which the kernel will assign it a free port.

Comment (zz-jason) [reply store/helper/helper_test.go]: Got it, thanks.

**Pull request merged**